---
title: "Transcriptomics and the analysis of RNA-Seq data"
author: "Rachel Ng"
date: "February 20, 2020"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Import today's data
Glucocorticoids inhibit inflammatory processes and are often used to treat asthma because of their anti-inflammatory effects on airway smooth muscle (ASM) cells. 
SNPs in CRISPLD2 in previous GWAS associated with inhaled corticosteroid resistance/bronchodilator response in asthma patients
Today, our starting point is a count matrix: each cell indicates number of reads originating from a particular gene (in rows) for each sample (in columns)

```{r}
counts <- read.csv("airway_scaledcounts.csv", stringsAsFactors = F)
metadata <- read.csv("airway_metadata.csv", stringsAsFactors = F)
```


##Ultimately, we are asking: Does the drug change transcription?

How many genes are we dealing with?
```{r}
nrow(counts)
```

We want to calculate the average counter per gene for control samples and for the drug treated samples.Then, we will compare these averages to see if there is a significant difference.

Now I want to access the control only data in the metadata
```{r}
#give you the IDs of the control samples
metadata$id[metadata$dex == "control"]
#metadata[metadata$dex == "control", ]$id

#define control as control samples
control <- metadata[metadata[,"dex"]=="control",]
control

```

Now use the $id of the control data to access control columns in counts and calc average
```{r}
control.mean <- rowSums(counts[ ,control$id]/nrow(control))
```
### Now for the treated samples...
```{r}
treated <- metadata[metadata$dex == "treated" , ]
treated

treated.mean <- rowSums (counts[, treated$id] ) /nrow(treated)
```

We will combine our meancount for bookeeping purposes.
```{r}
meancounts <- data.frame(control.mean, treated.mean)
meancounts
```

Let's plot
```{r}
plot(meancounts)
```

Now we'll plot by log
```{r}
plot(log2(meancounts))
```

Make new column in meancounts called "Log2FC"
```{r}
meancounts$log2fc <- log2(meancounts[,"treated.mean"]/meancounts[,"control.mean"])
head(meancounts)

#note: log(0) will return NA
```

Remove any gene that has zero avg vounts
```{r}
zero.vals <- which(meancounts[,1:2]==0, arr.ind=TRUE)

to.rm <- unique(zero.vals[,1])
mycounts <- meancounts[-to.rm,]
head(mycounts)
```

```{r}
nrow(mycounts)
```


A common threshold used for calling something differentially expressed is a log2(FoldChange) of greater than 2 or less than -2. Let’s filter the dataset both ways to see how many genes are up or down-regulated.

```{r}
up.ind <- mycounts$log2fc > 2
down.ind <- mycounts$log2fc < (-2)

#number of upregulated and number of down regulated genes
sum(up.ind)
sum(down.ind)

```

Notes on: Fold change (log ratios)
- very small but stastically significant fold change might not be relevant to a cell's function
- of interest to know which genes are most dramatically regulated, as these are often thought to reflect change in biologically meaningful transcripts/pathways

#DESeq2 analysis
DESeq2 is an R package for analyzing count-based NGS data like RNA-seq.

```{r}
library(DESeq2)

```

Setting up the input for DESeq2
- design is set to what we are actually comparing, in this case 'dex'

```{r}
dds <- DESeqDataSetFromMatrix(countData=counts, 
                              colData=metadata, 
                              design=~dex, 
                              tidy=TRUE)
dds
```

```{r}
dds <- DESeq(dds)
```

```{r}
res <- results(dds)
```
Plotting
A volcano plot shows the log fold change on the X-axis, and the −log10 of the p-value on the Y-axis (the more significant the p-value, the larger the −log10 of that value will be
```{r}
plot(res$log2FoldChange, res$padj)
```

```{r}

plot(res$log2FoldChange, -log(res$padj))

# Add some cut-off lines
abline(v=c(-2,2), col="darkgray", lty=2)
abline(h=-log(0.1), col="darkgray", lty=2)

```

```{r}
# Setup our custom point color vector 
mycols <- rep("gray", nrow(res))
mycols[ abs(res$log2FoldChange) > 2 ]  <- "red" 

#Volcano plot with custom colors 
plot( res$log2FoldChange,  -log(res$padj), 
 col=mycols, ylab="-Log(P-value)", xlab="Log2(FoldChange)" )

#v is vertical; h is horizontal
abline(v=c(-2,2), col="gray", lty=2)
abline(h=-log(0.1), col="gray", lty=2)
```

