---
title: "class18: cancer genomics datasets""
author: "Rachel Ng"
date: "3/5/2020"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Connect to the NCI-GDC database

The NCI-GDC (cancer genomics commons) is a major portal for cancer genomics data.

Here we use the bioconductor package **GenomicDataCommons** to talk to this database and download data etc. all from R

```{r}
library(GenomicDataCommons)
library(TCGAbiolinks)
library(maftools)
```

##Querying the GDC from R

The pipe operator can be useful in R and someitmes is more clear to read.
```{r}
c(1:10) %>% mean() == mean(c(1:10))
```

Note that the **facet()** and **aggregations()** functions here are from the GenomicDataCommons package and act to group all cases by the project id and then count them up.

```{r}
cases_by_project <- cases() %>%
  facet("project.project_id") %>%
  aggregations()
head(cases_by_project)
```

### Plot this overview data
```{r}
x<- cases_by_project$project.project_id
head(x)
barplot(log(x$doc_count), names.arg = x$key, las = 2)

#las = 2 turns x-axis labels vertical
```

## Variant analysis with R
Read in the MAF file we found
```{r}
maf.file <- GDCquery_Maf(tumor="PAAD", pipelines = "mutect")
head(maf.file)
```

```{r}
vars = read.maf(maf = maf.file, verbose = FALSE)
```

```{r}
plotmafSummary(vars)
```


## Oncoplots

```{r}
oncostrip(maf=vars, genes=c("KRAS", "TP53"))
```

Another plot focusing on KRAS in our particular dataset.
```{r}
lollipopPlot(vars, gene='KRAS')
```
```{r}
lollipopPlot(vars, gene='TP53')
```

#Part 2: designing a personalized vaccination
```{r}
library(bio3d)
seqs <- read.fasta("lecture18_sequences.fa")
```

Using the conserv function:
`conserv()` Quantifies residue conservation in a given protein sequence alignment by calculating the degree of amino acid variability in each column of the alignment.
Finding where residue conservation variablity is less than 1 and storing it in mutant.sites

```{r}
## Calculate positional identity scores (non-identicals)
ide <- conserv(seqs$ali, method="identity")
mutant.sites <- which(ide < 1) 
mutant.sites
```
We are including gap positions here and this is not what we want! So we need to exclude the gap positions:

```{r}
##exclude gap positions from analysis
gaps <- gap.inspect(seqs)
gaps$row

##column number of gaps
gaps$t.inds

##positions with no gaps (non-gaps)
gaps$f.inds
```

Find the overlap between mutant.sites and non-gaps
```{r}
mutant.sites <- mutant.sites[mutant.sites %in% gaps$f.inds]
mutant.sites
```

Find the -8 before and -8 after (i.e. all the sequence segment containign all possible 9-mer sequences w tumor specific amino acids)

```{r}
start <- mutant.sites - 8
end <- mutant.sites + 8

```


```{r}
## Make a "names" label for our output sequences (one per mutant)
mutant.names <- paste0(seqs$ali["P53_wt",mutant.sites],
                       mutant.sites,
                       seqs$ali["P53_mutant",mutant.sites])

# Blank matrix to store sub-sequences
storeseq <- matrix("-", nrow=length(mutant.sites), ncol=17)
rownames(storeseq) <- mutant.names

## Extract each sub-sequence
for(i in 1:length(mutant.sites)) {
  storeseq[i,] <- seqs$ali["P53_mutant",start[i]:end[i]]
}

storeseq
```

```{r}
## Output a FASTA file for further analysis
write.fasta(seqs=storeseq, ids=mutant.names, file="subsequences.fa")
```

